### PR TITLE
refactor: use signals in user components

### PR DESCRIPTION
**Pull request by tobi-or-not-tobi**
Refactor address mixin and move to signals

closes [HRZ-2986](https://spryker.atlassian.net/browse/HRZ-2986)

[HRZ-2986]: https://spryker.atlassian.net/browse/HRZ-2986?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: libs/domain/user/address-list/src/address-list.component.ts [hunk 1]
```diff
@@ -22,39 +16,37 @@ import { styles } from './address-list.styles';
...
 
-  @asyncState()
-  protected addresses = valueType(resolve(AddressService).getAddresses());
+  protected add = signal(resolve(AddressService).getAddresses());
 
   @state()
   protected selectedAddressId?: string;
 
   protected willUpdate(changedProperties: PropertyValues): void {
+    const addresses = this.addresses();
```
Comment (dunqan) [inline libs/domain/user/address-list/src/address-list.component.ts]: Did you consider to move it to an effect? Or it didn't work?

File: libs/domain/user/src/mixins/address-component.mixin.ts [hunk 2]
```diff
@@ -34,30 +27,18 @@ export const AddressMixin = <
...
-
-    @observe()
-    protected addressId$ = new BehaviorSubject(this.addressId);
-
-    @property({ type: Object }) address?: Address;
-
-    @observe('address')
-    protected addressProp$ = new BehaviorSubject(this.address);
+    @signalProperty() addressId?: string;
+    @signalProperty() addressData?: Address;
```
Comment (dunqan) [inline libs/domain/user/src/mixins/address-component.mixin.ts]: I think attribute name is more important than usable property (as it lands in DOM), and we don't have this convention of `somethingData`. Not saying it's wrong, but maybe worth reconsidering...

File: libs/domain/user/address-list/src/address-list.component.ts (hunk 1 above)
Comment (tobi-or-not-tobi) [reply libs/domain/user/address-list/src/address-list.component.ts]: have not considered, good point

File: libs/platform/experience/src/mixins/content.mixin.ts [hunk 3]
```diff
@@ -21,10 +21,24 @@ export declare class ContentMixinInterface<OptionsType, ContentType>
 
   protected data: { options: OptionsType; content: ContentType };
 
+  /**
+   * @deprecated
```
Comment (dunqan) [inline libs/platform/experience/src/mixins/content.mixin.ts:25]: I assume we want to drop those soon? (when finishing refactor I guess) As ideally we won't have deprecared in 1.0.

File: libs/domain/user/address/src/address.component.ts [hunk 4]
```diff
@@ -20,29 +19,20 @@ export class AddressComponent extends AddressMixin(
...
-      }
+    if (multiline)
+      return this.renderMultiline(schema ?? defaultMultilineSchema, address);
 
-      return html`${this.formatLine(
-        options.schema ?? defaultSinglelineSchema,
-        address
-      )}`;
-    })}`;
+    return this.formatLine(schema ?? defaultSinglelineSchema, address);
```
Comment (dunqan) [inline libs/domain/user/address/src/address.component.ts]: this one could be implemented as computed, but not a big deal

**Review by dunqan — DISMISSED** [review]

File: libs/platform/experience/src/mixins/content.mixin.ts (hunk 3 above)
Comment (tobi-or-not-tobi) [reply libs/platform/experience/src/mixins/content.mixin.ts:25]: For sure. I just want a mechanism during the refactors to highlight these thigns

**Review by dunqan — APPROVED** [review]

**Pull request merged**